### DISCUSSION TITLE
`dev` errors in serverless functions after upgrade to Vercel CLI 19.1.1

### DISCUSSION THREAD

**Discussion by darekrossman** [vercel/vercel#4724] [Help] [answered]
We've been using `vercel dev` v19.0.1 for awhile without any issues (and loving it). After upgrading to v19.1.1 we've started getting an error around the instantiation of a third-party class.

```
Class constructor RemoteGraphQLDataSource cannot be invoked without 'new'
```

The offending code in `/api/graphql.js`:
```
import {ApolloGateway, RemoteGraphQLDataSource} from '@apollo/gateway'

class ServiceDataSource extends RemoteGraphQLDataSource {
 // omitted for brevity
}

const handler = (req, res) => {
  const new ApolloGateway({
      buildService({name, url}) {
        return new ServiceDataSource({url}) // <-- this is where the issue happens
      },
    })

   // omitted for brevity
}
```

Im still trying to pin down what exactly changed between CLI versions. It seems like a transpilation issue with extending a third-party class and IM not sure how to work around it. Taking out the line `new ServiceDataSource()` solves the issue, but thats as far as Ive been able to narrow it down. 

Ive seen a few similar discussions on the Apollo github, but every solution is centered around modifying webpack or babel in the context of something like Next. However, I couldnt find much info about how the underlying vercel build process actually works, or how to configure around it.

**Comment by TooTallNate** [comment, marked as answer]
Thank you for bringing this up! I have identified the problem and [created a pull request](https://github.com/vercel/vercel/pull/4728) that fixes the issue.
